pkg/cpuallocator/allocator.go

```diff
@@ -943,6 +966,21 @@ func (c *cpuPriorities) cmpCPUSet(csetA, csetB cpuset.CPUSet, prefer CPUPriority
 		return 0
 	}
 
+	// For low prio request, favor cpuset with the tightest fit.
+	if cpuCnt > 0 && prefer == PriorityLow {
+		prefA := csetA.Intersection(c[prefer]).Size()
+		prefB := csetB.Intersection(c[prefer]).Size()
+		if prefA >= cpuCnt || prefB >= cpuCnt {
+			if prefB == 0 {
+				return -1
+			}
+			if prefA == 0 {
+				return 1
+			}
```

Comment:
What do we need this special cases for? Why would the abs(return value) need to be 1 in theses cases?

Reply:
Uhmm... if either one is zero (set has no proper CPUs) we need to favor the other one. But in all other cases we favor the tighter fit (the one with fewer but sufficient number of proper CPUs), not the loser one. But there was a snafu there (the '== 0' special cases should have been '< cpuCnt' ones) if the whole test was to be written like that. Anyway, I rewrote it now to be hopefully both correct and easier to understand.